Naming convention doc (#822)
nsidc/earthaccess: docs/contributing/naming-convention.md

# Contributing Naming Convention

## General Guidelines

This document outlines the naming conventions for our project. These conventions are intended to promote consistency and clarity across different components.

### Specific Conventions

#### Python Code

  **Style**: We follow the official Python Enhancement Proposal 8 [PEP8]((https://peps.python.org/pep-0008/#package-and-module-names) ) guidelines for naming conventions.

#### Jupyter Notebooks

>  **File Names**: File names should start with an underscore (_). Please use underscores instead of hyphens.

Reviewer: ```suggestion
* **File names**: Please use underscores (`_`) instead of hyphens (`-`).
```

I don't think we should have an underscore prefix. Could you share your rationale?

Author: Oh, I made a mistake there. I wanted to write exactly what you wrote :)

Reviewer: :+1: :)

#### Documentation

>  **Directory and File Names**: Directory and file names should use hyphens (-) instead of underscores.

Reviewer: ```suggestion
* **Directory and file names**: Directory names and Markdown file names should use hyphens (`-`) instead of underscores (`_`).
```

#### Tests

> 1. **Test File and Function Names**: Test files and functions should start with `test_`.

Reviewer: ```suggestion
1. **Test file and member names**: Test files and methods should start with `test_`. Classes should start with `Test`.

    !!! note "Test File Naming Convention"
        We do not use the _test.py suffix for test files

```

Is "member" the right terminology? :shrug: 

I'd like this note to be associated with list item number 1, but I'm not sure if this will format correctly, can you test it out?

2. **Additional Guidelines**: For additional guidelines, please refer to the [pytest documentation](https://docs.pytest.org/en/stable/explanation/goodpractices.html#conventions-for-python-test-discovery).

> 
> !!! note "Test File Naming Convention"
>     We do not use the _test.py suffix for test files

Reviewer: ```suggestion
```
